aws_lambda_powertools/metrics/base.py

```diff
@@ -412,73 +331,19 @@ def decorate(event, context):
                 if capture_cold_start_metric:
                     self._add_cold_start_metric(context=context)
             finally:
-                self.flush_metrics(raise_on_empty_metrics=raise_on_empty_metrics)
+                if not raise_on_empty_metrics and not self.metric_set:
+                    warnings.warn(
+                        "No application metrics to publish. The cold-start metric may be published if enabled. "
+                        "If application metrics should never be empty, consider using 'raise_on_empty_metrics'",
+                        stacklevel=2,
+                    )
+                else:
+                    self.flush_metrics(raise_on_empty_metrics=raise_on_empty_metrics)
```

Comment:
```suggestion
                self.flush_metrics(raise_on_empty_metrics=raise_on_empty_metrics)
```

This `if block` is already implemented in `flush_metrics` method.